src/FSharp.Data.GraphQL.Server/ReflectionHelper.fs

```diff
@@ -140,6 +144,50 @@ module internal ReflectionHelper =
 
     let isPrameterMandatory = not << isParameterOptional
 
+    let unwrapOptions (ty : Type) =
+        if ty.FullName.StartsWith OptionTypeName || ty.FullName.StartsWith ValueOptionTypeName then
+            ty.GetGenericArguments().[0]
+        else ty
+
+    let isAssignableWithUnwrap (from: Type) (``to``: Type) =
+
+        let checkCollections (from: Type) (``to``: Type) =
+            if
+                // TODO: Implement support of other types of collections using collection initializers
+                (``to``.FullName.StartsWith ListTypeName || ``to``.FullName.StartsWith ArrayTypeName)
+                && (from.IsGenericType
```

Comment:
Should you also check `to.IsGenericType` because you use it below as taking the [0] argument. Similarly in unwrapOptions above.... it's hard to find the right balance of "how much type-checks will take time" vs "how the code works on edge-cases (when misused by a later PR)"

Reply:
Yep, before this PR @dim-34 just implemented some corner case I was not thought about 🙂
As I checked that it is `Option`1` I suppose I already know it is generic